### DISCUSSION TITLE
v1.4

### DISCUSSION THREAD

**Discussion by eduardofilo** [eduardofilo/RG350_adam_image#178] [Announcements]
## Installation instructions

* [In english](https://github.com/eduardofilo/RG350_adam_image/blob/master/README.md): It is mandatory to follow the instructions or the console will not boot up.
* [En español](https://github.com/eduardofilo/RG350_adam_image/blob/master/README_es.md): Es obligatorio seguir las instrucciones o la consola no arrancará.

## Upgrading instructions

Check [this guide](https://github.com/eduardofilo/RG350_adam_image/#upgrade) in README.

## Change log
* Support for GCW-Zero integrated in PGv1 image. From now on, this image also includes a script for selecting the console model. So like in main image, it is mandatory to follow the installation instructions in [README](https://github.com/eduardofilo/RG350_adam_image/blob/master/README.md) or the console will not boot up.
* ODbeta updated to 2021-12-12 thanks to @pcercuei and all the collaborators of the [project](https://github.com/OpenDingux/linux). Surely it solves issue #22.
* RetroArch updated to 2022-02-03.
    * New cores:
        * a5200_libretro.so: Atari 5200. Replaces atari800_libretro in OPK wrapper.
        * vitaquake2_libretro.so (plus three more for mission packs): Quake2.
        * jaxe_libretro.so: CHIP-8 (not in SimpleMenu so use from GMenu2X).
        * bk_libretro.so: Elektronika - BK-0010/BK-0011 (not in SimpleMenu so use from GMenu2X; needs BIOS).
        * freechaf_libretro.so: Fairchild ChannelF (not in SimpleMenu so use from GMenu2X).
        * minivmac_libretro.so: Apple Macintosh (not in SimpleMenu so use from GMenu2X;  needs BIOS).
    * Configuration changes:
        * Left stick as D-pad in all RA cores (issue #67).
        * Changed "WHDLoad Support" RA option in Amiga (discussion #136).
        * Swapped order of cap32 and crocods Amstrad CPC RetroArch cores because cap32 now has sound glitches.
        * Replaced 2022-02-03 dosbox_pure_libretro.so with 2021-10-13 version due to compatibility problems with the newer one.
* SimpleMenu updated to 10.3. Some configuration changes:
    * Fixed some extensions (.bin for Intellivision and .g64 for Commodore 64) (issue #66).
    * New system Super Game Boy (discussion #122).
    * New system Quake2 based in 4 new RA cores for base game and three mission packs.
    * New theme [BetaMax](https://github.com/eduardofilo/BetaMax-SimpleMenu-theme) thanks to @Yarkiebrown.
    * Fixed SimpleMenu overclocking configuration (issue #175).
    * Changed configuration of ColecoVision to use the new gearcoleco_libretro.so RA core as default. It needs BIOS so it remains the standalone one that works without it.
    * Changed configuration of Quake to fit [supported games](https://docs.libretro.com/library/tyrquake/#loading-quake-and-expansion-paks) (mission packs).
    * In Quake and Quake2 is applied a trick described in [README](https://github.com/eduardofilo/RG350_adam_image/#quake) thanks to @neilswann80.
* FBA updated to [this version](https://github.com/plrguez/fba-sdl/releases/tag/r19_2021-10-18-.44_od_beta-fix-06) thanks to @plrguez.
* PCSX4All updated to [last version](https://github.com/jdgleaver/RG350_pcsx4all/releases/tag/1.5) thanks to @jdgleaver (issue #175).
* Installed uae4all Amiga standalone emulator thanks to @goldmojo.
* Replaced ScummVM standalone emulator by [upstreamed version](https://downloads.scummvm.org/frs/scummvm/) thanks mainly to @citral23. See special instructions for this emulator in [README](https://github.com/eduardofilo/RG350_adam_image/#standalone).
* Installed [this music player](https://github.com/plrguez/gmu-odbeta/releases/tag/v0.10.2-rc1-odbeta-20211024) thanks to @plrguez (issue #41).

<hr /><em>This discussion was created from the release <a href='https://github.com/eduardofilo/RG350_adam_image/releases/tag/v1.4'>v1.4</a>.</em>

**Comment by valzi** [comment]
Does the Simplemenu update allow access to Cannonball?

**Reply by eduardofilo** [reply]
If the game is copied to the `apps` directory of the external car, it is internally in the `games` category and does not require selecting a file to be launched, it should appear and work from the `Games` section of SimpleMenu.
I have tested with this version and it is.
https://github.com/SeongGino/RetroGame350-AppRepo/raw/master/Games-Ports/cannonball.opk

You do need to add datafiles to the `~/.cannonball/roms/` directory though.

**Reply by valzi** [reply]
I finally tried this and it worked for all of my games except cannonball.opk. Should I create an issue? I believe this is a SimpleMenu problem rather than something we can fix, but I have no idea.

**Reply by eduardofilo** [reply]
Did you add the game datafiles?

**Comment by valzi** [comment]
Oh, and thank you for your effort! I am so happy this exists.

**Comment by Medard22** [comment]
Thank you for yet another great release. May I have a suggestion for changes in default retroarch settings for some handheld cores? In my opinion they just don´t look good and it´s a hassle to change settings for all cores with every release.

GameBoy and GameBoy Color - Keep Aspect Ratio on by default (they just look horrible when stretched).
GameGear - Integer Scale off by default (on RG350M it makes game window small when on), Grid2x filter instead of Grid3xDarken filter (the Grid3xDarken filter causes game colors to look very washed out on all systems - black becomes grey)
Lynx - Grid2x instead of Grid3xDarken
WonderSwan -  Keep Aspect Ratio on by default, Grid2x instead of Grid3xDarken

Thanky you and other devs for keeping our portables up-to-date :)

**Reply by eduardofilo** [reply]
Thank you.
The problem is that the image is designed for multiple screens which made me reach a difficult compromise. In the RA installer I do keep different settings for high and low resolution screens, but with the image I can't. I could consider making different files to restore with Py Backup for each console with those configs, but it is a job that never ends because there are different criteria for it. In general, for example, I prefer pixel perfect at the cost of losing some screen surface, even on small screens like the RG280, but I have already seen that in general this criterion is not the most accepted.

**Reply by Medard22** [reply]
It's not a big deal, and changing it is simple. Retroarch includes a lot of options that new users could find intimidating. I believe that at the very least, the base configuration should maintain the original system aspect ratio. Beginners may find it difficult to understand why games are stretched. Particularly in systems with a nearly 1:1 aspect ratio, such as the GameBoy, or systems with a wide aspect ratio, such as the Lynx and WonderSwan. However, this is just my opinion :)

**Comment by Medard22** [comment]
After quick testing I have further feedback. 
As mentioned in #120 base Quake games crashes early on in the section right after the bridge. 
I managed to run base Quake2 game, but the expansions don´t work. I copied the pak0.pak files from my Quake2 GOG version into the respective expansion folders, but they wont´t launch. Vitaquake2 doesn´t have much of a documentation. Anyone else had luck?

**Reply by eduardofilo** [reply]
Have you selected the corresponding core?
Check this:
https://github.com/eduardofilo/RG350_adam_image/#quake2

**Reply by Medard22** [reply]
No, I didn´t. Now I see it, my bad.  "You need to set the corresponding OPK in SimpleMenu or open the appropriate one..." Oops :D It works once properly set up.

**Comment by swamperjmb** [comment]
All good updates! Since this is a mish-mash of Open Dingy and RetroArch, is there a way to add snaps?

**Reply by eduardofilo** [reply]
By `snaps` do you mean the ROM previews in SimpleMenu?

**Reply by swamperjmb** [reply]
Video snap's. I first thought I could use video snap's as I added them to
the previews folder. I may have done something wrong but it only seems to
work with png files, not video files.  Maybe I forgot to add the
gamelist.xl file.
[…](https://github.com/eduardofilo/RG350_adam_image/discussions/178#)

**Reply by SuperBadger77** [reply]
Pretty sure Simple Menu doesn’t support videos. You could request it as a feature on the GitHub. It’s just roms, with pictures with a matching file name in png format in the .previews folder.

**Reply by swamperjmb** [reply]
I'm not familiar with all the systems these handhelds have. I'm very familiar with retropie.  I did see where the rg351v plays video snaps but it runs on EmuElec, not sure it has simple menu.  Sorry, just curious if it's even possible, I'm good if it's not, I'm more than happy how everything works now.

**Reply by eduardofilo** [reply]
There is a version of EmulationStation for this system, but is old and buggy. Other frontends like PyMenu supported a special video snap format, but it hasn't evolved much since two years:
https://github.com/JackD83/PyMenu/releases

In general, unless someone compiles EmulationStation for ODbeta, it won't be easy to see that functionality.

**Comment by Derek85** [comment]
Hello. 

Any plans in the future to increase compatibility with custom DOOM Wads and Conversions? The options for PRDoom is extremely limited. In my opinion that's the only thing missing from making my RG 280 V a perfect system.

**Reply by neilswann80** [reply]
I think this is more of a limitation with Retroarch/Libretro development for OpenDingux Beta, than it is for developemnt towards "Adam Image".

For ODBeta we're limited to the cores available here:
https://buildbot.libretro.com/nightly/dingux/mips32-odbeta/latest/

**Reply by Derek85** [reply]
There has to be a way to add GZ Doom to increase compatibility.

**Reply by neilswann80** [reply]
If someone ports it to ODBeta that would be a start.... But I don't think it has been yet???  I thought the only ODBeta Doom ports were PRBoom, FreeDOOM and Odamex;  of which PRBoom is the most compatible across the retro handsets and supporting mod wads... and actually has a retroarch core so can be neatly merged into the Adam Image.

**Reply by eduardofilo** [reply]
Well, as @neilswann80 says, in the image I limit myself to keeping an installation of ODb, RA and SimpleMenu updated. It is true that an almost unlimited number of ports and standalone emulators can be added, but the design of the image is to limit myself as much as possible to RA to contain the size of the project. It is always up to users to add what they are missing for themselves. Trying to please everyone and ensure that the image does not become untenable is a difficult commitment.

**Reply by Derek85** [reply]
@eduardofilo That's fair feedback. It is a nearly perfect image in my opinion. Hopefully this becomes a feature later.

**Comment by zugurudumba** [comment]
Thank you for the update!

Is there an issue if I use the fresh install method instead of upgrading from the previous version? I don't mind having everything wiped.

**Reply by neilswann80** [reply]
There isn't an upgrade option for v1.4, so you'll need to do the fresh install/flash any way.  This should only affect the data on your primary microsd card (and you can backup savegames and configs via PYBackup), your secondary (game roms) microsd will be unaffected.

**Comment by K9-711** [comment]
Hi. Just flashed Adam 1.4 (coming from 1.3). I have to adjust my favorites.sav file. Otherwise simplemenu didn't start. Actually I'm done with it. But the issue is that some games are listed twice in favorites while one entry shows the device on the right top like Neo Geo Pocket and the second entry only shows an X. 

First I added for each category which had issues a game into favorites and compared the line with my original favorit.sav file from 1.3. Then accordingly I edited the affected lines. But still, something is wrong.

Any suggestions?

For instance this game appears twice under favorites:

FINALBURN ALPHA;/media/sdcard/roms/FBA/ghouls.zip;Ghouls'n Ghosts (World);/media/data/apps/;fba-multi.opk;0;/media/sdcard/roms/FBA/;996

Edit: I figured it out. There mist not be empty lines between. Now all good.

**Comment by pavanini** [comment]
no se si a alguien mas le pasa, pero después de flashear se crea la carpeta de QUAKE2 en la micro sd externa pero esta solo da mensaje de error, no se deja borrar y si se intenta entrar a ella en la consola, esta se buguea

**Reply by eduardofilo** [reply]
Se tienen que crear no sólo esa carpeta, sino todas las que hay en el siguiente listado:
https://github.com/eduardofilo/RG350_adam_image/blob/master/README_es.md#roms

Sí que son especiales Quake y Quake2 en el sentido de que aparecerán en SimpleMenu aunque no haya ficheros de esos juegos instalados. Es por el truco que se aplicó en esos dos casos para resolver el problema de que todos los ficheros se llaman igual y no hay forma si no de distinguir los juegos. Está explicado aquí:
https://github.com/eduardofilo/RG350_adam_image/blob/master/README_es.md#quake

Lo que te tiene que ocurrir al abrirlos es que RetroArch se cerrará inmediatamente volviendo a SimpleMenu al no encontrar los ficheros (.pak) de esos juegos que tienes que instalar si quieres jugarlos.

Es cierto que tal y como monté el sistema, no hay forma de ocultar esos dos sistemas como no sea editando el fichero `home and computers.ino` y borrando las entradas `QUAKE` y `QUAKE2` en la primera línea de ese fichero. Pensaré algo para ver si se puede mejorar eso.

**Comment by Chris3911** [comment]
Hello guys I was testing this version the 1.4 on the 280v and so far is good but I noticed something since it's boot up I noticed like a problem with the screen like flickering that never happened before really bad I will try to take a picture of it and maybe I can share with you guys somewhere

**Reply by eduardofilo** [reply]
Oh. I work on the image mainly on an RG280V and I haven't noticed anything. Only when I put the data cable the brightness goes up a bit.

**Reply by Chris3911** [reply]
That's weird.. but for real I'm having this trouble maybe it's just in my device who knows. but this doesn't happen in the 1.3 at all so basically what I have to do is restart the device and then it's fixed but if I navigate too much through simple menu this happened super weird.

**Reply by eduardofilo** [reply]
Surely the screen control has changed in the latest version of ODb which is the one installed in 1.4 and it affects your machine more than others. In any case, I can't do much more than update ODb in the future in the hope that it improves that situation.

**Reply by Chris3911** [reply]
For now it's fine. Definitely I can wait no problem. This annoying me for a bit but not a big deal. I can definitely play and that's the important.

**Reply by pcercuei** [reply]
Nope, nothing changed recently.

**Reply by eduardofilo** [reply]
In that case @Chris3911, it has to be a coincidence that your console showed a change when moving to 1.4. It could also be that the brightness level has changed between both versions (I didn't take note if I did) and the new level produces that effect in your device. In that case it could be solved simply by modifying it. Make tests.

**Reply by neilswann80** [reply]
@Chris3911 
If you lower you screen brightness until the screen turns off (PWR and LEFT-DPAD), then increase the screen brightness to maximum (PWR and RIGHT-DPAD), then return the brightness to your preferred level... does that clear the flickering issue for you?

**Reply by Chris3911** [reply]
If I play with the brightness  works correctly so far. And for the screen flickering I haven't see that again. that happened a lot on the first and the second boot when I first installed the 1.4 but right now it seems to works fine I will let you know guys if something

**Reply by Chris3911** [reply]
> In that case @Chris3911, it has to be a coincidence that your console showed a change when moving to 1.4. It could also be that the brightness level has changed between both versions (I didn't take note if I did) and the new level produces that effect in your device. In that case it could be solved simply by modifying it. Make tests.

How do I modify that? its a txt file ?

**Reply by eduardofilo** [reply]
I basically meant that you try different brightness levels and see if some of them flicker and not others. It would also be nice to install 1.3 and do the same tests to see if there are any differences, but if you say it only happened to you a few isolated times, I'd leave it at that.

**Reply by Chris3911** [reply]
Yes so far nothing maybe in the future will happen again I will try to record it . It will be easy for me because one start flickering it won't stop until I reboot I will send a picture of a similar thing

**Reply by Chris3911** [reply]
![Screenshot_20220212-123635_Chrome_1.png](https://user-images.githubusercontent.com/99370210/153721949-79021975-a8d0-424d-a857-e42d44a077ee.png)

**Reply by neilswann80** [reply]
Wow... that's more than the flickering I was imagining.  It's taking me back to my rather psychedelic youth!

That looks like it might be hardware to me.  Anyone else concur?

**Reply by Chris3911** [reply]
Similar to neogeo emu when booting a rom

**Reply by Chris3911** [reply]
@eduardofilo are you latin?

**Reply by eduardofilo** [reply]
But that photo is of a laptop screen, right?

I was born in Spain.

**Reply by Chris3911** [reply]
> But that photo is of a laptop screen, right?
> 
> I was born in Spain.
 Yes sir it is a laptop but it's just an example


By the way to you know the language? Sabes hablar español?

**Reply by eduardofilo** [reply]
De hecho hablar hablar, es lo único que hablo :-)

**Reply by Chris3911** [reply]
> De hecho hablar hablar, es lo único que hablo :-)

Ah pues bien me caes súper bien no sabías que hablabas el idioma Yo también hablo español de hecho es mi idioma nativo

**Reply by eduardofilo** [reply]
Encantado de saludarte pues.
Pásate por mi sitio de apuntes. Está en español y hay mucha información sobre las consolas que soporta la imagen:
https://apuntes.eduardofilo.es/retro-emulacion/rg-350.html
Y también artículos sueltos en el blog:
https://apuntes.eduardofilo.es/index.html

**Reply by Chris3911** [reply]
> Encantado de saludarte pues.
> Pásate por mi sitio de apuntes. Está en español y hay mucha información sobre las consolas que soporta la imagen:
> https://apuntes.eduardofilo.es/retro-emulacion/rg-350.html
> Y también artículos sueltos en el blog:
> https://apuntes.eduardofilo.es/index.html

Wow qué bueno brutal eres el más duro en realidad si no fuera por ti no fuera tan asombrosa la consola Gracias por todo tu esfuerzo y dedicación en realidad me caes muy bien tienes telegram a WhatsApp?

**Reply by eduardofilo** [reply]
Bueno gracias, pero sólo recopilo lo que otros hacen.

**Reply by Chris3911** [reply]
> Bueno gracias, pero sólo recopilo lo que otros hacen.

Sea lo que sea que haces es asombroso me gustaría hablar contigo más pero no aquí porque está básicamente hecho para comentarios no para hablar tienes alguna red social o algún sitio donde pueda comunicarte?

**Reply by eduardofilo** [reply]
Discord: eduardofilo#7650

**Reply by Chris3911** [reply]
> Discord: eduardofilo#7650

Vale ya te mandé una solicitud
